src/lib.rs

```diff
@@ -481,6 +481,43 @@ impl<T: Debug + PrimInt + One + Zero> Vob<T> {
         }
     }
 
+    ///Returns contents of Vob as Vec of bytes.
+    ///
+    ///For a Vob of length n, the first bit will be the 0th element,
+    ///the last bit would be the (n-1)th element.
+    ///
+    /// # Examples
+    /// ```
+    ///use vob::Vob;
+    ///let mut x = Vob::from_elem(5, false);
+    ///x.set(2, true);
+    ///x.set(4, true);
+    ///assert_eq!(x.to_bytes(), [0b00101000]);
+    ///```
+    pub fn to_bytes(&self) -> Vec<u8> {
+        fn bit<T: Debug + One + PrimInt + Zero>(vob: &Vob<T>, byte: usize, bit: usize) -> u8 {
+            let offset = byte * 8 + bit;
+            if offset >= vob.len {
+                0
+            } else {
+                (vob[offset] as u8) << (7 - bit)
+            }
+        }
+
+        let len = self.len / 8 +
+                  if self.len % 8 == 0 { 0 } else { 1 };
+        (0..len).map(|i|
+            bit(self, i, 0) |
+            bit(self, i, 1) |
+            bit(self, i, 2) |
+            bit(self, i, 3) |
+            bit(self, i, 4) |
+            bit(self, i, 5) |
+            bit(self, i, 6) |
+            bit(self, i, 7)
```

Comment:
This can be done much more efficiently. It is possibly to use `reverse` on nightly.

Reply:
You could probably do `iter_set_bits` to grab a more efficient slice as well.

Reply:
Hi, I've pushed changes that are ready to be reviewed